packages/core/src/components/slider/rangeSlider.tsx

```diff
@@ -55,7 +55,11 @@ export class RangeSlider extends AbstractPureComponent2<IRangeSliderProps> {
         const { value, ...props } = this.props;
         return (
             <MultiSlider {...props}>
-                <MultiSlider.Handle value={value[RangeIndex.START]} type="start" intentAfter={Intent.PRIMARY} />
+                <MultiSlider.Handle
+                    value={value[RangeIndex.START]}
+                    type="start"
+                    intentAfter={props.intent || Intent.PRIMARY}
```

Comment:
we try to avoid implicit type coercion like this. instead, please set the default intent to `Intent.PRIMARY` inside `public static defaultProps`. then this line can just be `={props.intent}`

Reply:
this change has been made